src/core/openthread.cpp

```diff
@@ -929,20 +929,22 @@ bool otIsSingleton(void)
     return mEnabled && sThreadNetif->GetMle().IsSingleton();
 }
 
-ThreadError otActiveScan(uint32_t aScanChannels, uint16_t aScanDuration, otHandleActiveScanResult aCallback)
+ThreadError otActiveScan(uint32_t aScanChannels, uint16_t aScanDuration, otHandleActiveScanResult aCallback,
+                         void *aCallbackContext)
 {
     return sThreadNetif->GetMac().ActiveScan(aScanChannels, aScanDuration, &HandleActiveScanResult,
-                                             reinterpret_cast<void *>(aCallback));
+                                             reinterpret_cast<void *>(aCallback),
```

Comment:
I'd prefer if we do not push the multi-layer callback into the Mac::ActiveScan() API.  Can we keep state about the API's client in the API layer?